### PR TITLE
users without a name break selection of some 2fa keys

### PR DESCRIPTION
**Pull request by terencehonles**
For some 2FA keys both the user [1] `name` and `displayName` are required. If a user's name is not set, this provides the user's username as both the `name` and the `displayName`.

This can be tested and confirmed by setting up [the flask demo app](https://github.com/duo-labs/py_webauthn#flask-demo) and applying the patch to remove the `displayName` validation:

```diff
diff --git a/flask_demo/app.py b/flask_demo/app.py
index c0083fd..d0aa284 100644
--- a/flask_demo/app.py
+++ b/flask_demo/app.py
@@ -59,12 +59,12 @@ def index():
 def webauthn_begin_activate():
     # MakeCredentialOptions
     username = request.form.get('register_username')
-    display_name = request.form.get('register_display_name')
+    display_name = request.form.get('register_display_name') or ''
 
     if not util.validate_username(username):
         return make_response(jsonify({'fail': 'Invalid username.'}), 401)
-    if not util.validate_display_name(display_name):
-        return make_response(jsonify({'fail': 'Invalid display name.'}), 401)
+#    if not util.validate_display_name(display_name):
+#        return make_response(jsonify({'fail': 'Invalid display name.'}), 401)
 
     if User.query.filter_by(username=username).first():
         return make_response(jsonify({'fail': 'User already exists.'}), 401)
```

And then using a Yubi 5 nano will not work (`ykman info`): 

```
Device type: YubiKey 5 Nano
Serial number: [redacted]
Firmware version: 5.1.2
Form factor: Nano (USB-A)
Enabled USB interfaces: FIDO

Applications
OTP     	Disabled	
FIDO U2F	Enabled 	
OpenPGP 	Disabled	
PIV     	Disabled	
OATH    	Disabled	
FIDO2   	Enabled 
```

[1]: https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredentialCreationOptions/user

### CODE REVIEW

**Comment by terencehonles** [comment]
Alternatively it might make sense to warn the user that they need to set their name.

**Comment by di** [comment]
Hi @terencehonles! This makes sense to me. Could you update the tests as well?

@woodruffw would you mind taking a quick look?

**Comment by woodruffw** [comment]
Taking a look now!

**Comment by woodruffw** [comment]
This looks reasonable to me. We could also change the WebAuthn user generation to only use `username` for the `displayName`, since that shouldn't affect any existing credentials. But the fixed behavior looks fine, and provides a graceful degradation.

**Comment by terencehonles** [comment]
> Hi @terencehonles! This makes sense to me. Could you update the tests as well?

I _think_ this should cover it. I removed fail-fast from the github workflow since I'm having some issues running `make test` locally and I would like it to try to run all the tests, but it looks like there might be some other issue going on that's affecting all the tests.

**Comment by terencehonles** [comment]
Looks like a `docker-compose up db` took care of my issue since I probably jumped too quickly to running the tests and assumed the make recipe would take care of making sure the db was started.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ on: [push, pull_request]
 jobs:
   test:
     strategy:
+      fail-fast: false
```
Comment (di) [inline .github/workflows/ci.yml:6]:
```suggestion
```

**Comment by di** [comment]
Thanks!

**Pull request merged**